src/include/btree.i

```diff
@@ -556,9 +557,16 @@ __wt_page_only_modify_set(WT_SESSION_IMPL *session, WT_PAGE *page)
             page->modify->first_dirty_txn = last_running;
     }
 
-    /* Check if this is the largest transaction ID to update the page. */
+    /* Check if this is the largest transaction to update the page. */
     if (WT_TXNID_LT(page->modify->update_txn, session->txn->id))
         page->modify->update_txn = session->txn->id;
+
+    /*
+     * Assume the update will be get a timestamp at least as large as the largest durable timestamp
+     * we've seen.
+     */
+    if (page->modify->update_ts < (durable_ts = S2C(session)->txn_global.durable_timestamp))
```

Comment:
Fair call, I've reverted the changes to LRU policy.

fwiw, I don't expect this to be a noticeable slowdown: we're already tracking `update_txn` in the same path.  That said, the benefit is minor (only a few percent size saving in my tests).  An alternative approach that didn't consider pages until `update_ts` is stable unless eviction was under some pressure was a little more effective but seemed more risky.  I still couldn't get the index size down near the "optimal" size where no time windows were saved.

w.r.t. statistics, I think the ones @sulabhM added that track how many pages are written with each type of time information are valuable.  It might help to add the number of bytes written for time windows -- if that becomes large compared with `cache_bytes_write`, that would be telling.  Also interesting would be tracking the bytes of underfill -- how many bytes are wasted due to writes being rounded up to the allocation size?

Reply:
I think Alex G. and Haseeb wrote those statistics. Should we create a ticket for any more of those ?

Reply:
@sulabhM @michaelcahill, I have an open PR for fixing some related stats. I can either add the stats there or I can create a new ticket and I'll use that.

Reply:
Thanks @hbokhari, either makes sense to me: if you already have a ticket for adding some stats, feel free to add these ones as well.